sql/tests: TestRandomSyntaxSQLSmith failed [unable to encode table key: *tree.DTSVector]

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13713799?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13713799?buildTab=artifacts#/) on master @ [b59a83a1cd31d8e00f39cf32b1cf99b41b96e28e](https://github.com/cockroachdb/cockroach/commits/b59a83a1cd31d8e00f39cf32b1cf99b41b96e28e):

Random syntax error:

```
    rsg_test.go:910: Crash detected: server panic: pq: internal error: unable to encode table key: *tree.DTSVector
```
Query:

```
        WITH
        	with_9050 ("col _52520")
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						(
        							('''DQVm'':188A,381C,456C,993B ''HBJeAKhfBR'' ''IiqQ'' ''KxyNkAxoN'':240A ''UIHOF'' ''YuhdtLkwwi'':211C ''Z'':188B,688C ''k'' ''uusmZZ'' ''xtZ''':::TSVECTOR, 'a8da54ec-adf4-4cc9-809a-2ce13fe9fcf2':::UUID)
        						),
        						(
        							('''BpPmCRaaTP'':218A,923A ''hkIMbPKOvI'':60A,894B ''nzAlcgtZnA'':102B,490B ''v'':142B,173A,764B ''zDUnJneHE''':::TSVECTOR, 'fc28804d-a344-4573-ba44-00be6976aec3':::UUID)
        						),
        						(
        							('''CVjP'' ''DKqCzzGX'' ''LkeLYtempk'':75B ''Y'':598A ''llvObNpuZQ''':::TSVECTOR, '31bc2969-f650-4635-8957-969f57502c34':::UUID)
        						)
        				)
        					AS tab_22008 ("col _52520")
        		),
        	withͫ_9051 ("col̐ _52522")
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						(2556969473:::OID),
        						(679083309:::OID),
        						(
        							CASE
        							WHEN NULL THEN NULL
        							ELSE (
        								SELECT
        									tab_22009.tableoid AS col_52521
        								FROM
        									defaultdb.public."table%p_1"@"table%p_1_col1__12_c'ol1__0_expr_idx" AS tab_22009
        								WHERE
        									tab_22009.col1_̘_5
        								ORDER BY
        									tab_22009.crdb_internal_mvcc_timestamp ASC NULLS FIRST,
        									tab_22009.col1__9 NULLS FIRST
        								LIMIT
        									1:::INT8
        							)
        							END
        						)
        				)
        					AS tab_22010 ("col̐ _52522")
        		)
        SELECT
        	('''UfzaPyBd'' ''aAVP'':322C,894B ''bjQeP'':370C,928B ''fqlbvV'' ''zYeMC'':326C,589C,999C':::TSVECTOR, '00000000-0000-0000-0000-000000000000':::UUID)
        		AS col_52523
        FROM
        	with_9050 AS c̘te_ref_2581
        GROUP BY
        	c̘te_ref_2581."col _52520"
        HAVING
        	every(true::BOOL)::BOOL;
```
Schema:

```
    rsg_test.go:719: To reproduce, use schema:
    rsg_test.go:721: SET sql_safe_updates = false;;
    rsg_test.go:721: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:721: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:721: CREATE TABLE "table%p_1" ("c'ol1__0" BIT(41) NOT NULL, col1_͖_1 GEOMETRY NOT NULL, "col\\u82F41__2" TIMESTAMPTZ NOT NULL, col1__3 REGTYPE NULL, "col 1__4" REGCLASS NULL, col1_̘_5 BOOL, "col 1__6" TIME NOT NULL, "col1_ _7" GEOGRAPHY NOT NULL, col1__8 CHAR NOT NULL, col1__9 STRING NULL AS (CASE WHEN "col 1__4" IS NULL THEN e'5\x0f!h\x19\x12I!':::STRING ELSE '{3_':::STRING END) VIRTUAL, col1__10 STRING AS (lower(CAST(col1_̘_5 AS STRING))) STORED, "c ol1__11" STRING NULL AS (CASE WHEN col1__3 IS NULL THEN '-_P_':::STRING ELSE e'V\r{\x15m\\':::STRING END) STORED, col1__12 STRING NOT NULL AS (lower(CAST("col 1__6" AS STRING))) VIRTUAL, PRIMARY KEY ("c'ol1__0" ASC), UNIQUE (col1__9) STORING (col1__3, col1_̘_5, "col 1__6", col1__8, col1__10, "c ol1__11"), INDEX (col1_̘_5, col1__10) STORING ("col 1__6", "c ol1__11"), INDEX (col1__12 ASC, "c'ol1__0" ASC, lower(CAST(col1_͖_1 AS STRING))) STORING (col1_͖_1, col1__3, "col 1__4", "col 1__6", "col1_ _7", col1__10), INVERTED INDEX ("col 1__4" ASC, col1__10 gin_trgm_ops) WHERE (("table%p_1".col1__9 > '"':::STRING) OR "table%p_1".col1_̘_5) OR ("table%p_1".col1__8 > '"':::STRING), INVERTED INDEX (col1__10 gin_trgm_ops) WHERE (((((("table%p_1".col1__8 >= 'X':::STRING) AND ("table%p_1"."col 1__6" > '00:00:00':::TIME)) OR ("table%p_1".col1__9 > e'\x00':::STRING)) AND (NOT "table%p_1".col1_̘_5)) OR ("table%p_1"."col\\u82F41__2" < '3000-01-01 00:00:00+00':::TIMESTAMPTZ)) OR ("table%p_1"."c ol1__11" >= '':::STRING)) AND ("table%p_1".col1__12 != '':::STRING));
    rsg_test.go:721: CREATE TABLE tablé_2 (col2__0 PG_LSN NOT NULL, "col""2'__1" REGCLASS, ")col2__2" REGROLE NULL, col2__3 INTERVAL NULL, "col2_ \\U0000E9A8_4" INT4 NOT NULL, col2__5 TSVECTOR NOT NULL, col2_̵_6 FLOAT8 NOT NULL, col2__7 STRING NULL, col2__8 BYTES, col2_😳_9 TIMESTAMP[] NOT NULL, co😵l2__10 "char" NOT NULL, "cOl2_|_11" UUID NULL, "col%p2__12" REGCLASS NULL, col2__13 INTERVAL NOT NULL, "co l2
_ _14" GEOMETRY NOT NULL, "cOl2__15" INT4 NOT NULL, " col2_̹_16" STRING NOT NULL AS (lower(CAST("cOl2_|_11" AS STRING))) STORED, "col2
        __17" STRING NOT NULL AS (lower(CAST(col2__5 AS STRING))) STORED, PRIMARY KEY (" col2_̹_16" DESC, col2_̵_6, "col2_ \\U0000E9A8_4" ASC), INVERTED INDEX (co😵l2__10 gin_trgm_ops) WHERE ((((((tablé_2.co😵l2__10 <= e'\'':::STRING) AND (tablé_2." col2_̹_16" = e'\x00':::STRING)) OR (tablé_2.col2__7 > '':::STRING)) AND (tablé_2."col2_ \\U0000E9A8_4" >= 1:::INT8)) AND (tablé_2."col2
        __17" < 'X':::STRING)) OR (tablé_2."cOl2__15" != 2147483647:::INT8)) AND (tablé_2.col2__8 >= '\x':::BYTES), INDEX (")col2__2" ASC, "col2_ \\U0000E9A8_4", "col2
        __17" ASC), INDEX (col2_̵_6) WHERE (((((((tablé_2.col2__7 <= e'\'':::STRING) AND (tablé_2.co😵l2__10 >= e'\U00002603':::STRING)) OR (tablé_2."cOl2__15" > 0:::INT8)) OR (tablé_2.col2_̵_6 <= 3.4028234663852886e+38:::FLOAT8)) AND (tablé_2.col2__8 <= '\xff':::BYTES)) AND (tablé_2." col2_̹_16" >= e'\'':::STRING)) OR (tablé_2."col2_ \\U0000E9A8_4" > 32767:::INT8)) AND (tablé_2."col2
        __17" = '"':::STRING), FAMILY (col2__3), FAMILY ("col2_ \\U0000E9A8_4", "cOl2_|_11"), FAMILY ("cOl2__15"), FAMILY (col2_̵_6), FAMILY (col2__5, col2__0), FAMILY (col2__13), FAMILY ("col2
        __17", col2__7, ")col2__2"), FAMILY (" col2_̹_16"), FAMILY ("co l2
_ _14"), FAMILY (col2__8, "col""2'__1", col2_😳_9, "col%p2__12", co😵l2__10));
    rsg_test.go:721: CREATE TABLE "tablE_3" (co̊l3__0 TIMETZ NULL, "col?3__1" GEOMETRY, col3__2 STRING COLLATE de_DE, col3__3 INT4 NOT NULL, col3__4 STRING NOT NULL AS (lower(CAST(co̊l3__0 AS STRING))) STORED, col3__5 STRING AS (lower(CAST(col3__2 AS STRING))) STORED, PRIMARY KEY (col3__4 ASC, col3__3 ASC), UNIQUE (col3__3 DESC, col3__2 DESC), UNIQUE ((col3__3 + 950067361:::INT8) ASC) STORING (col3__5) WHERE (("tablE_3".col3__4 >= 'X':::STRING) OR ("tablE_3".col3__3 != 0:::INT8)) OR ("tablE_3".co̊l3__0 <= '24:00:00-15:59':::TIMETZ), UNIQUE (col3__5), UNIQUE (lower(CAST("col?3__1" AS STRING)) ASC, col3__2 ASC, co̊l3__0), INDEX (co̊l3__0 ASC, col3__3 DESC) STORING ("col?3__1", col3__2) WHERE ("tablE_3".col3__4 != e'\x00':::STRING) AND ("tablE_3".co̊l3__0 < '24:00:00-15:59':::TIMETZ), UNIQUE (col3__5 DESC) STORING ("col?3__1") WHERE ((("tablE_3".col3__5 >= '"':::STRING) OR ("tablE_3".col3__3 >= 2147483647:::INT8)) OR ("tablE_3".col3__4 >= '':::STRING)) OR ("tablE_3".co̊l3__0 <= '24:00:00-15:59':::TIMETZ), INDEX (col3__5) WHERE ((("tablE_3".col3__3 = 1:::INT8) OR ("tablE_3".col3__5 < e'\x00':::STRING)) AND ("tablE_3".col3__4 < 'X':::STRING)) OR ("tablE_3".co̊l3__0 > '00:00:00+15:59':::TIMETZ));
    rsg_test.go:721: ALTER TABLE "table%p_1" INJECT STATISTICS e'[{"avg_size": 22, "columns": ["col1__3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1504313058374368047, "histo_col_type": "", "name": "__auto__", "null_count": 246552372809621407, "row_count": 6234812524253154073}, {"avg_size": 2, "columns": ["\\"c\'ol1__0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4442131066867911831, "histo_buckets": [{"distinct_range": 0, "num_eq": 2129432981504189872, "num_range": 0, "upper_bound": "01111111111111111111111111111111111111111"}], "histo_col_type": "BIT(41)", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 6234812524253154073}, {"avg_size": 23, "columns": ["col1__8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3421876178886699272, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6234812524253154073}, {"avg_size": 6, "columns": ["col1__9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2440323196883047161, "histo_buckets": [{"distinct_range": 0, "num_eq": 6779859447187351876, "num_range": 0, "upper_bound": "\'g"}, {"distinct_range": 600000000, "num_eq": 50000000, "num_range": 600000000, "upper_bound": "\\u00033}p\\u001c\\u0005\\t\\u000fP"}, {"distinct_range": 30000000000, "num_eq": 30000000000, "num_range": 30000000000, "upper_bound": "m\\u0018=_\\u0003YM\\u001aG"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 495430671165118052, "row_count": 6234812524253154073}, {"avg_size": 2, "columns": ["\\"c ol1__11\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4098050344746015698, "histo_col_type": "", "name": "__auto__", "null_count": 2684509059495211503, "row_count": 6234812524253154073}, {"avg_size": 22, "columns": ["col1_\u0356_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4010267685794112167, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6234812524253154073}, {"avg_size": 14, "columns": ["\\"col 1__4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2401492726627130523, "histo_buckets": [{"distinct_range": 0, "num_eq": 8056633577666688914, "num_range": 0, "upper_bound": "0"}, {"distinct_range": 6420.604762269394, "num_eq": 497797480561858823, "num_range": 9000, "upper_bound": "3705248078"}, {"distinct_range": 473397.7584036621, "num_eq": 2465722182053927914, "num_range": 2000000, "upper_bound": "1661778005"}, {"distinct_range": 0, "num_eq": 4127895264314001524, "num_range": 8581076118118000176, "upper_bound": "1796051632"}, {"distinct_range": 70000, "num_eq": 0, "num_range": 70000, "upper_bound": "516930929"}, {"distinct_range": 0, "num_eq": 8852350739452502990, "num_range": 300, "upper_bound": "1182716792"}, {"distinct_range": 452621863.6102029, "num_eq": 8000, "num_range": 800000000, "upper_bound": "2270438588"}], "histo_col_type": "REGCLASS", "histo_version": 3, "name": "__auto__", "null_count": 732518383583219473, "row_count": 6234812524253154073}, {"avg_size": 23, "columns": ["col1_\u0318_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4355665225198288945, "histo_col_type": "BOOL", "histo_version": 3, "name": "__auto__", "null_count": 6133975629910130887, "row_count": 6234812524253154073}, {"avg_size": 10, "columns": ["\\"col 1__6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3952174141492929470, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6234812524253154073}, {"avg_size": 30, "columns": ["col1__10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2428326261880347877, "histo_col_type": "", "name": "__auto__", "null_count": 479684928641979471, "row_count": 6234812524253154073}, {"avg_size": 7, "columns": ["col1__12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1783946754263221361, "histo_buckets": [{"distinct_range": 0, "num_eq": 6418276235868908855, "num_range": 0, "upper_bound": "\\u001ax\\\\"}, {"distinct_range": 1237859.1349946682, "num_eq": 30000, "num_range": 3000000, "upper_bound": ";\\u0017y"}, {"distinct_range": 0, "num_eq": 3000000, "num_range": 4814602333804607345, "upper_bound": "[4f"}, {"distinct_range": 5919521.7420879705, "num_eq": 9111570670725772069, "num_range": 100000000, "upper_bound": "\\u0016@A4>\\u001c\\u000fTe"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 6234812524253154073}, {"avg_size": 28, "columns": ["\\"col\\\\\\\\u82F41__2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 537219077537162460, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6234812524253154073}, {"avg_size": 29, "columns": ["\\"col1_ _7\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1826432782146082013, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6234812524253154073}]':::JSONB;
    rsg_test.go:721: ALTER TABLE tablé_2 INJECT STATISTICS e'[{"avg_size": 0, "columns": ["\\"col2_ \\\\\\\\U0000E9A8_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2__7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2__8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"co l2\\r_ _14\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\" col2_\u0339_16\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 3000, "num_range": 0, "upper_bound": "\\""}, {"distinct_range": 1768306918818361000, "num_eq": 0, "num_range": 1768306918818361042, "upper_bound": "\\u0002N"}, {"distinct_range": 124141966159738620, "num_eq": 500000, "num_range": 638913319763818294, "upper_bound": "&c."}, {"distinct_range": 0, "num_eq": 5000000000, "num_range": 0, "upper_bound": "E\\u0018\\u001f"}, {"distinct_range": 0, "num_eq": 9000000000, "num_range": 3724969447096788317, "upper_bound": "slA"}, {"distinct_range": 2662910361746623500, "num_eq": 929487199058781552, "num_range": 4518880386075879079, "upper_bound": "\\u001b}Ip\\u0015\\u0010\\u0017"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"col\\"\\"2\'__1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2__3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"col2\\n__17\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2__5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2_\u0335_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 9000, "num_range": 0, "upper_bound": "0.9777874141721288"}, {"distinct_range": 0, "num_eq": 400000000, "num_range": 3000, "upper_bound": "-0.10355240364561813"}], "histo_col_type": "FLOAT8", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2_\U0001F633_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"cOl2_|_11\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"cOl2__15\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2__0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\")col2__2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 10000, "num_range": 0, "upper_bound": "623841302"}, {"distinct_range": 3000, "num_eq": 3540387866660222959, "num_range": 3000, "upper_bound": "2050342552"}, {"distinct_range": 0, "num_eq": 1349789041496302161, "num_range": 3556506564335284865, "upper_bound": "4074732321"}, {"distinct_range": 120.53560516923999, "num_eq": 4219679958468389517, "num_range": 300, "upper_bound": "1445062247"}, {"distinct_range": 600000000, "num_eq": 90000, "num_range": 600000000, "upper_bound": "488252328"}, {"distinct_range": 50000000000, "num_eq": 8000, "num_range": 50000000000, "upper_bound": "984002161"}, {"distinct_range": 1211754724204619000, "num_eq": 5696629455864719609, "num_range": 1744024349927795994, "upper_bound": "2758451188"}, {"distinct_range": 6684876467999843000, "num_eq": 2755499275050920928, "num_range": 6684876467999843782, "upper_bound": "1559393143"}, {"distinct_range": 326252963.3136655, "num_eq": 200, "num_range": 800000000, "upper_bound": "2034821373"}], "histo_col_type": "REGROLE", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["co\U0001F635l2__10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 30000, "num_range": 0, "upper_bound": "\\u0000"}, {"distinct_range": 7782273421.102408, "num_eq": 6384802911886662146, "num_range": 8000000000, "upper_bound": "\\u0015"}, {"distinct_range": 0, "num_eq": 800, "num_range": 40000000, "upper_bound": "\'"}, {"distinct_range": 5399980662641417000, "num_eq": 90000000000, "num_range": 5399980662641416784, "upper_bound": "{"}], "histo_col_type": "\\"char\\"", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"col%p2__12\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col2__13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}]':::JSONB;
    rsg_test.go:721: ALTER TABLE "tablE_3" INJECT STATISTICS e'[{"avg_size": 14, "columns": ["co\u030Al3__0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 69, "histo_col_type": "", "name": "__auto__", "null_count": 592, "row_count": 700}, {"avg_size": 16, "columns": ["\\"col?3__1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 274, "histo_col_type": "", "name": "__auto__", "null_count": 625, "row_count": 700}, {"avg_size": 18, "columns": ["col3__2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 75, "histo_col_type": "", "name": "__auto__", "null_count": 635, "row_count": 700}, {"avg_size": 27, "columns": ["col3__3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 236, "histo_buckets": [{"distinct_range": 0, "num_eq": 3466902180714412931, "num_range": 0, "upper_bound": "1638873801"}], "histo_col_type": "INT4", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 700}, {"avg_size": 4, "columns": ["col3__4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 694, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 700}, {"avg_size": 25, "columns": ["col3__5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 398, "histo_buckets": [{"distinct_range": 0, "num_eq": 3777451426231467338, "num_range": 0, "upper_bound": ""}, {"distinct_range": 0, "num_eq": 1021227861701182708, "num_range": 0, "upper_bound": "s:"}, {"distinct_range": 7743194099644075000, "num_eq": 5659437865126482882, "num_range": 8442012639087445362, "upper_bound": "cxx"}, {"distinct_range": 732566263203834100, "num_eq": 7000000, "num_range": 1136997890911544162, "upper_bound": "X}\\u0019<]{*o"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 105, "row_count": 700}]':::JSONB;
    rsg_test.go:721: CREATE TYPE rand_typ_0 AS ENUM ('vubjbj', 'hpntrh', 't', 'hsnyxd', 'anxh', 'dr');
    rsg_test.go:721: CREATE TYPE rand_typ_1 AS ENUM ('ccpb', 'trfy', 'okubor', 'tzq');
    rsg_test.go:721: CREATE TYPE rand_typ_2 AS ENUM ('ydy', 'jrb', 'tmfkq', 'j', 'bpnv', 'v');
    rsg_test.go:721: SET sql_safe_updates = false;;
    rsg_test.go:721: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:721: ALTER TABLE seed SPLIT AT VALUES (8044336415234281275);
    rsg_test.go:721: ALTER TABLE seed SPLIT AT VALUES (6591941016335691210);
    rsg_test.go:721: ALTER TABLE seed SCATTER;
    rsg_test.go:723: 
    rsg_test.go:724: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith1648679997
--- FAIL: TestRandomSyntaxSQLSmith (376.18s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35701

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13731870?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13731870?buildTab=artifacts#/) on master @ [0baf22a03d5f55e2611701bc723e3e0b713ab051](https://github.com/cockroachdb/cockroach/commits/0baf22a03d5f55e2611701bc723e3e0b713ab051):

Random syntax error:

```
    rsg_test.go:910: Crash detected: server panic: pq: internal error: expected equivalence dependants to be its closure: (47)-->(46), (35)==(46), (46)==(35), (27)==(47), (47)==(27), (47)-->(51), (47)==(49), (49)==(47) (3)
```
Query:

```
        WITH
        	"wit
h_4352" (col_24330)
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						(NULL),
        						(ARRAY[4194245226:::OID,3230689084:::OID,1344639712:::OID]),
        						(ARRAY[1184563187:::OID,3519674121:::OID]),
        						(ARRAY[2445577561:::OID,0:::OID,3669984343:::OID,0:::OID]),
        						(ARRAY[621708786:::OID,4154750271:::OID,1710616223:::OID,2429091596:::OID]),
        						(ARRAY[574175311:::OID,3830019171:::OID])
        				)
        					AS "t%vab"" _10460" (col_24330)
        		),
        	with_4353 (col_24331) AS (SELECT * FROM (VALUES ((-18428):::INT8)) AS tab_10461 (col_24331))
        SELECT
        	merge_stats_metadata(tab_10463._jsonb::JSONB) OVER (PARTITION BY tab_10463._jsonb ORDER BY tab_10463._uuid NULLS FIRST, tab_10462._decimal DESC NULLS FIRST)::JSONB
        		AS col_24332,
        	variance(3.658203721870504108E+36:::DECIMAL::INT8)::DECIMAL AS col_24333,
        	'0870db5c-4401-4676-9e12-bf976f7d7905':::UUID AS "\\x53col _24334"
        FROM
        	defaultdb.public.seed@seed__int8__float8__date_idx AS tab_10462,
        	defaultdb.public.seed@[0] AS tab_10463
        	JOIN defaultdb.public.table_3@table_3_col3__1_idx AS tab_10464 ON
        			(tab_10463._string) = (tab_10464.col3__1)
        			AND (tab_10463.tableoid) = (tab_10464.tableoid)
        			AND (tab_10463._float4) = (tab_10464.col3__2)
        			AND (tab_10463._float8) = (tab_10464.col3__4)
        GROUP BY
        	tab_10463._jsonb, tab_10463._uuid, tab_10462._decimal
        LIMIT
        	98:::INT8;
```
Schema:

```
    rsg_test.go:719: To reproduce, use schema:
    rsg_test.go:721: SET sql_safe_updates = false;;
    rsg_test.go:721: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:721: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:721: CREATE TABLE table_1 ("col͒1_"" _0" BOX2D NOT NULL, "c|o l1__1" REGROLE, col1__2 REGCLASS, """col1__3" DECIMAL NOT NULL, col1__4 NAME NULL, "col1_%q_5" INT4 NOT NULL, co😯l1__6 FLOAT8 NOT NULL, "c!o😕L1_ _7" NAME NOT NULL, "c
        ol1__8" "char" NOT NULL, col1__9 STRING COLLATE en_US NOT NULL, col1__10 STRING NULL, "c😖ol%v1__11" DATE, col1__12 DECIMAL NOT NULL, col1__13 BYTES, "c ol1__14" FLOAT8 NOT NULL AS (co😯l1__6 + 0.9936464997449823:::FLOAT8) VIRTUAL, col1__15 DECIMAL NOT NULL AS ("""col1__3" + col1__12) VIRTUAL, PRIMARY KEY (col1__9), INDEX (col1__9, col1__12 ASC, (col1__12 + """col1__3")), INDEX (col1__12 DESC) WHERE ((((((((((table_1.col1__10 <= e'\x00':::STRING) OR (table_1.col1__15 = (-1.0):::DECIMAL)) AND (table_1."c ol1__14" != '-Inf':::FLOAT8)) OR (table_1.col1__12 > (-1.0):::DECIMAL)) OR (table_1."c😖ol%v1__11" < '1970-01-01':::DATE)) OR (table_1."col1_%q_5" >= (-32768):::INT8)) AND (table_1."""col1__3" <= 1:::DECIMAL)) OR (table_1.col1__13 < '\x22':::BYTES)) OR (table_1.col1__4 <= '':::STRING:::NAME)) AND (table_1.co😯l1__6 != '-Inf':::FLOAT8)) AND (table_1."c!o😕L1_ _7" >= '':::STRING:::NAME) VISIBILITY 0.53, FAMILY (co😯l1__6), FAMILY ("c|o l1__1"), FAMILY ("col1_%q_5"), FAMILY (col1__4), FAMILY (col1__12, col1__10, "c!o😕L1_ _7", col1__2), FAMILY ("c😖ol%v1__11"), FAMILY (col1__13, "col͒1_"" _0", "c
        ol1__8", col1__9, """col1__3"));
    rsg_test.go:721: CREATE TABLE table_2 ("co%eel2__0" BIT(27) NOT NULL, "col2%v_&_1" BYTES NULL, col2__2 TIMESTAMP NOT NULL, "col""2__3" OID NULL, PRIMARY KEY ("co%eel2__0" DESC), INDEX ("co%eel2__0" ASC, "col""2__3", "col2%v_&_1" DESC) STORING (col2__2) WHERE (table_2."col2%v_&_1" <= '\x00':::BYTES) AND (table_2.col2__2 = '3000-01-01 00:00:00':::TIMESTAMP) NOT VISIBLE, INDEX ("co%eel2__0") WHERE (table_2.col2__2 = '4714-11-24 00:00:00 BC':::TIMESTAMP) OR (table_2."col2%v_&_1" >= '\x':::BYTES), INDEX ("col2%v_&_1" ASC, col2__2 DESC) STORING ("col""2__3") VISIBILITY 0.32, UNIQUE ("col""2__3" ASC) WHERE (table_2.col2__2 < '0001-01-01 00:00:00':::TIMESTAMP) AND (table_2."col2%v_&_1" != '\x27':::BYTES), UNIQUE ("col2%v_&_1" ASC, "co%eel2__0", lower(CAST("col""2__3" AS STRING)) DESC) STORING ("col""2__3"), INDEX (col2__2 DESC) STORING ("col""2__3"), UNIQUE (col2__2 DESC, "col""2__3" DESC) STORING ("col2%v_&_1") WHERE (table_2.col2__2 != '294276-12-31 23:59:59.999999':::TIMESTAMP) OR (table_2."col2%v_&_1" = '\x58':::BYTES), UNIQUE ("col2%v_&_1", col2__2 ASC) WHERE table_2.col2__2 >= '0001-01-01 00:00:00':::TIMESTAMP, INDEX ("co%eel2__0" DESC, col2__2 DESC) WHERE (table_2.col2__2 < '4714-11-24 00:00:00 BC':::TIMESTAMP) OR (table_2."col2%v_&_1" >= '\xe29883':::BYTES));
    rsg_test.go:721: CREATE TABLE table_3 ("col%p3__0" REFCURSOR, col3__1 VARCHAR NOT NULL, col3__2 FLOAT8 NOT NULL, "&%vcol3_
        _3" STRING NOT NULL AS (lower(col3__1)) VIRTUAL, col3__4 FLOAT8 AS (col3__2 + 0.0:::FLOAT8) STORED, PRIMARY KEY (col3__2 DESC), INDEX (col3__1 ASC), UNIQUE (col3__4 ASC, col3__1 DESC) WHERE (table_3.col3__2 <= 'NaN':::FLOAT8) OR (table_3.col3__4 >= 'NaN':::FLOAT8), UNIQUE (col3__2, "&%vcol3_
        _3" DESC, col3__4 ASC) STORING ("col%p3__0") WHERE (((table_3."&%vcol3_
        _3" < 'X':::STRING) AND (table_3.col3__1 != '':::STRING)) AND (table_3.col3__4 < 'NaN':::FLOAT8)) AND (table_3.col3__2 <= (-1.0):::FLOAT8));
    rsg_test.go:721: ALTER TABLE table_1 INJECT STATISTICS e'[{"avg_size": 14, "columns": ["col1__12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2803014004624452580, "histo_buckets": [{"distinct_range": 0, "num_eq": 7288530526389750275, "num_range": 0, "upper_bound": "0.4930641601562689554"}, {"distinct_range": 6287522710518010000, "num_eq": 0, "num_range": 6287522710518010182, "upper_bound": "-19.75630266923052961"}], "histo_col_type": "DECIMAL", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 20, "columns": ["col1__15"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3299585153936269664, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 28, "columns": ["\\"col\u03521_\\"\\" _0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2172478420139609352, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 3, "columns": ["\\"c\\nol1__8\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1208805629747824111, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 11, "columns": ["\\"\\"\\"col1__3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2584378819178133855, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 17, "columns": ["co\U0001F62Fl1__6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3788030521476445971, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 18, "columns": ["\\"c!o\U0001F615L1_ _7\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2185844364322458497, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 7, "columns": ["col1__9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5023273508728184256, "histo_buckets": [{"distinct_range": 0, "num_eq": 20000000000, "num_range": 0, "upper_bound": "e\'\\\\U000EF7C8\' COLLATE en_US"}, {"distinct_range": 1083491719416842100, "num_eq": 900000, "num_range": 6209353817090035519, "upper_bound": "e\'\\\\U000018A9\\\\U000790FC\' COLLATE en_US"}, {"distinct_range": 0, "num_eq": 5179228101497784263, "num_range": 0, "upper_bound": "e\'\\\\U0005E7C5\\\\U000BDCA5\' COLLATE en_US"}, {"distinct_range": 0, "num_eq": 20000000, "num_range": 200000, "upper_bound": "e\'\\\\U000D59C8\\\\U0007EF0D\\\\U0005A89E\' COLLATE en_US"}, {"distinct_range": 4118899693649840600, "num_eq": 10000, "num_range": 4118899693649840776, "upper_bound": "e\'\\\\U000B1EF7\\\\U00024C62\\\\U000046F4\\\\U0008565A\\\\U000A4402\\\\U0004FEB9\' COLLATE en_US"}, {"distinct_range": 0, "num_eq": 8412051657315615228, "num_range": 0, "upper_bound": "e\'\\\\U000556D8\\\\U0005B3F6\\\\U0002FABC\\\\U0006155B\\\\U000273A0\\\\U000E16A5\' COLLATE en_US"}, {"distinct_range": 0, "num_eq": 70000, "num_range": 60000, "upper_bound": "e\'\\\\U000B94C5\\\\U00067813\\\\U00088A86\\\\U000A6AFD\\\\U0007BF61\\\\U00079F59\\\\U0000D2F4\\\\U00073359\' COLLATE en_US"}, {"distinct_range": 0, "num_eq": 3000000, "num_range": 0, "upper_bound": "e\'\\\\U00092B6E\\\\U00080C51\\\\U000B9974\\\\U000298D2\\\\U00037119\\\\U0009BC1E\\\\U000B1AB8\\\\U000EECA4\' COLLATE en_US"}, {"distinct_range": 40000000, "num_eq": 7000000, "num_range": 40000000, "upper_bound": "e\'\\\\U0009C5E7\\\\U0002D44D\\\\U00035717\\\\U0002859B\\\\U000750A4\\\\U00068AC1\\\\U0005E7E5\\\\U000DBFC4\' COLLATE en_US"}], "histo_col_type": "STRING COLLATE en_US", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 3, "columns": ["col1__10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4881016290659312411, "histo_col_type": "", "name": "__auto__", "null_count": 3562167950128534754, "row_count": 5128826996137067097}, {"avg_size": 17, "columns": ["col1__13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3083499627550478732, "histo_col_type": "", "name": "__auto__", "null_count": 1568827938654115038, "row_count": 5128826996137067097}, {"avg_size": 29, "columns": ["\\"c|o l1__1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3854263874401383301, "histo_col_type": "", "name": "__auto__", "null_count": 1566222314724022040, "row_count": 5128826996137067097}, {"avg_size": 9, "columns": ["col1__4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1163447834894985002, "histo_col_type": "", "name": "__auto__", "null_count": 2811029415819965691, "row_count": 5128826996137067097}, {"avg_size": 29, "columns": ["\\"col1_%q_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3971612750958835013, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 11, "columns": ["\\"c\U0001F616ol%v1__11\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1643893846418082118, "histo_col_type": "", "name": "__auto__", "null_count": 2760900762247046029, "row_count": 5128826996137067097}, {"avg_size": 26, "columns": ["\\"c ol1__14\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3274465312107721255, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5128826996137067097}, {"avg_size": 29, "columns": ["col1__2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2226176421076615751, "histo_col_type": "", "name": "__auto__", "null_count": 4091082432478701544, "row_count": 5128826996137067097}]':::JSONB;
    rsg_test.go:721: ALTER TABLE table_2 INJECT STATISTICS e'[{"avg_size": 10, "columns": ["\\"co%eel2__0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2917273290, "histo_buckets": [{"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": "010011011111111110110010111"}, {"distinct_range": 0, "num_eq": 8522674745288513864, "num_range": 3945993886354775678, "upper_bound": "011001111110111000001110101"}, {"distinct_range": 0, "num_eq": 30000, "num_range": 0, "upper_bound": "100111001011010011100101010"}, {"distinct_range": 0, "num_eq": 5000000000, "num_range": 90000000, "upper_bound": "111110110111110100110001110"}], "histo_col_type": "BIT(27)", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 4000000000}, {"avg_size": 12, "columns": ["\\"col2%v_&_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 880080046, "histo_buckets": [{"distinct_range": 0, "num_eq": 3468513170044587630, "num_range": 0, "upper_bound": "\\\\xff"}, {"distinct_range": 3548900670563194400, "num_eq": 20000, "num_range": 3548900670563194565, "upper_bound": "\\\\xba54e2"}, {"distinct_range": 0, "num_eq": 90000, "num_range": 0, "upper_bound": "\\\\x4425afdc3a"}, {"distinct_range": 0, "num_eq": 70000000, "num_range": 100000000000, "upper_bound": "\\\\x74e00b7a30a5"}, {"distinct_range": 0, "num_eq": 1000000, "num_range": 5984440984173849143, "upper_bound": "\\\\xddaa3f459a0c"}, {"distinct_range": 100000000, "num_eq": 2000000, "num_range": 100000000, "upper_bound": "\\\\x8901c675533b38"}, {"distinct_range": 52003295442723384, "num_eq": 1000000, "num_range": 93358258612279804, "upper_bound": "\\\\x4919ee5de1b26c79"}, {"distinct_range": 0, "num_eq": 206798835330123144, "num_range": 2821408484564141344, "upper_bound": "\\\\xd2bc1e5f2d91eb65"}], "histo_col_type": "BYTES", "histo_version": 3, "name": "__auto__", "null_count": 3652802723, "row_count": 4000000000}, {"avg_size": 4, "columns": ["col2__2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3644463900, "histo_buckets": [{"distinct_range": 0, "num_eq": 20000000000, "num_range": 0, "upper_bound": "2019-01-23 06:25:13.000856"}, {"distinct_range": 700, "num_eq": 5461547372379330620, "num_range": 700, "upper_bound": "1970-04-09 11:30:23.000949"}, {"distinct_range": 0, "num_eq": 4922144743791600156, "num_range": 4000000000, "upper_bound": "1983-06-15 05:15:05.000199"}, {"distinct_range": 940629.0795677841, "num_eq": 10000000000, "num_range": 3000000, "upper_bound": "1980-01-06 21:30:15.000543"}, {"distinct_range": 3689624642212803600, "num_eq": 305717989148904744, "num_range": 3689624642212803813, "upper_bound": "2018-12-03 09:18:47.000946"}], "histo_col_type": "TIMESTAMP", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 4000000000}, {"avg_size": 26, "columns": ["\\"col\\"\\"2__3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3416331775, "histo_buckets": [{"distinct_range": 0, "num_eq": 10, "num_range": 0, "upper_bound": "2448550350"}, {"distinct_range": 1183376733998935300, "num_eq": 9004521703557057436, "num_range": 5856465872311579399, "upper_bound": "4283037582"}, {"distinct_range": 0, "num_eq": 300000, "num_range": 8904576271217659435, "upper_bound": "2559928804"}, {"distinct_range": 1814340125356705500, "num_eq": 900000, "num_range": 1814340125356705646, "upper_bound": "3414505516"}, {"distinct_range": 0, "num_eq": 60000000000, "num_range": 10000, "upper_bound": "4107109422"}, {"distinct_range": 0, "num_eq": 900, "num_range": 344796989915393716, "upper_bound": "3999190958"}, {"distinct_range": 2112109748359648000, "num_eq": 8453862568378032698, "num_range": 5873838028357832144, "upper_bound": "2330499257"}, {"distinct_range": 2266945517460950500, "num_eq": 10000000000, "num_range": 2266945517460950589, "upper_bound": "751076348"}], "histo_col_type": "OID", "histo_version": 3, "name": "__auto__", "null_count": 3790456285, "row_count": 4000000000}]':::JSONB;
    rsg_test.go:721: ALTER TABLE table_3 INJECT STATISTICS e'[{"avg_size": 3, "columns": ["\\"col%p3__0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6250, "histo_col_type": "", "name": "__auto__", "null_count": 2423, "row_count": 7000}, {"avg_size": 20, "columns": ["col3__1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6163, "histo_buckets": [{"distinct_range": 0, "num_eq": 687640233045708970, "num_range": 0, "upper_bound": "\\""}, {"distinct_range": 969137356751713800, "num_eq": 400, "num_range": 3298962638408472680, "upper_bound": "\'"}, {"distinct_range": 1507.0617795664791, "num_eq": 1914026609620061663, "num_range": 3000, "upper_bound": "\\u000b-\\u000e"}, {"distinct_range": 2686362898465977300, "num_eq": 0, "num_range": 3115457088293869382, "upper_bound": "m%0"}, {"distinct_range": 60000, "num_eq": 700000000, "num_range": 60000, "upper_bound": "j\\u0016P\\u000e"}, {"distinct_range": 0, "num_eq": 758088453233637197, "num_range": 40000000, "upper_bound": "[\\u001dtn\\u0012"}, {"distinct_range": 0, "num_eq": 1598468452770686177, "num_range": 10000, "upper_bound": "\\u0002i.%\\u0005(\x7f"}, {"distinct_range": 0, "num_eq": 2000000000, "num_range": 1000000, "upper_bound": "\\u0011q\\u0014:Mku<"}], "histo_col_type": "VARCHAR", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 7000}, {"avg_size": 19, "columns": ["col3__2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1458, "histo_buckets": [{"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": "0.6670371443177597"}, {"distinct_range": 0, "num_eq": 5483258370484335519, "num_range": 594656871106202306, "upper_bound": "0.973270420082214"}, {"distinct_range": 2229104711957336600, "num_eq": 3736243423711704188, "num_range": 7390944951804598515, "upper_bound": "-0.22752407773196381"}, {"distinct_range": 5064023969188041000, "num_eq": 2347601593243736334, "num_range": 5439083896445982843, "upper_bound": "-0.30891126561270754"}, {"distinct_range": 630651571971804000, "num_eq": 40, "num_range": 853246826317794951, "upper_bound": "-0.9247691909578439"}, {"distinct_range": 214285388025631780, "num_eq": 0, "num_range": 599222027541678064, "upper_bound": "-1.922380307241682"}], "histo_col_type": "FLOAT8", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 7000}, {"avg_size": 25, "columns": ["\\"&%vcol3_\\n_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5359, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 7000}, {"avg_size": 28, "columns": ["col3__4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6579, "histo_buckets": [{"distinct_range": 0, "num_eq": 700000000, "num_range": 0, "upper_bound": "0.1505751633548229"}, {"distinct_range": 0, "num_eq": 100, "num_range": 80000000000, "upper_bound": "0.2652855219826106"}, {"distinct_range": 4290913.693465454, "num_eq": 60000, "num_range": 6000000, "upper_bound": "0.9008165873325502"}, {"distinct_range": 0, "num_eq": 1000, "num_range": 9000, "upper_bound": "-0.38053205531095546"}, {"distinct_range": 0, "num_eq": 9000, "num_range": 100000000000, "upper_bound": "-0.387902193152427"}], "histo_col_type": "FLOAT8", "histo_version": 3, "name": "__auto__", "null_count": 2706, "row_count": 7000}]':::JSONB;
    rsg_test.go:721: CREATE TYPE rand_typ_0 AS ENUM ('iklwdi', 'z', 'cuyqsb');
    rsg_test.go:721: SET sql_safe_updates = false;;
    rsg_test.go:721: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:721: ALTER TABLE seed SPLIT AT VALUES (2489279752014653004);
    rsg_test.go:721: ALTER TABLE seed SPLIT AT VALUES (6383416016364234637);
    rsg_test.go:721: ALTER TABLE seed SPLIT AT VALUES (4523893210943463069);
    rsg_test.go:721: ALTER TABLE seed SPLIT AT VALUES (6505220980128242171);
    rsg_test.go:721: ALTER TABLE seed SPLIT AT VALUES (3998339219981982317);
    rsg_test.go:721: ALTER TABLE seed SCATTER;
    rsg_test.go:723: 
    rsg_test.go:724: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith2204215493
--- FAIL: TestRandomSyntaxSQLSmith (377.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13776737?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13776737?buildTab=artifacts#/) on master @ [cc4fdffa8532d16544c48ef036689763f737dc6b](https://github.com/cockroachdb/cockroach/commits/cc4fdffa8532d16544c48ef036689763f737dc6b):

Random syntax error:

```
    rsg_test.go:909: Crash detected: server panic: pq: internal error: unable to encode table key: *tree.DTSQuery
```
Query:

```
        WITH
        	"w i😿th_7152" (col_39576, "c%9bol_39577", "%vcol_39578", "co%acl_39579")
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						(
        							'''jwrE'' <-> !''gOKDVyFQ'' <-> ''nKTdJZqLEh'' & ''qJzwRx''':::TSQUERY,
        							9223372036854775807:::INT8,
        							ARRAY[4293528709:::OID,3099633636:::OID],
        							'''CbsU'':478A,538B,588C,965A ''NvAlTjOZA'':210B ''WdTdc'':181A,274B ''a'':787A ''nEw'' ''nUVcHW'' ''objk'':272A,628B ''r'':905B,913A ''xv'':19B,56B,488C,901A':::TSVECTOR
        						),
        						(
        							NULL,
        							(-2832925099603743579):::INT8,
        							ARRAY[4231572989:::OID,3584439234:::OID,0:::OID],
        							'''nproURaAr'' ''uswoDzfWp''':::TSVECTOR
        						),
        						(NULL, NULL, ARRAY[3253684169:::OID], '''RpNJpiv'':251A,661B ''uXb''':::TSVECTOR),
        						(
        							'!''AbovO'' & !''Ei'' & ''OBXp'' | ''kOU'' | ''dSqvpmR''':::TSQUERY,
        							(-2680600367537064612):::INT8,
        							NULL,
        							'''HJdFgZ'' ''KLGwRgVq'' ''VEULwDUj'' ''YUrjK'' ''ZuemyJaxgA'' ''aujoKONFr'' ''ix'' ''liSQFo'' ''pjjjmkCN'':167A ''zhMfjcPSQe''':::TSVECTOR
        						),
        						(
        							'''DBilWAnk'' | ''O'' | ''ZahRomg'' | ''AeSM''':::TSQUERY,
        							NULL,
        							ARRAY[1477423181:::OID],
        							'''drX'':133A,961C ''eAMC'':292A,413B ''hjS'':171B,857C ''pnBo'':4C,247B,442C':::TSVECTOR
        						)
        				)
        					AS "?Ta
        b_17516" (col_39576, "c%9bol_39577", "%vcol_39578", "co%acl_39579")
        		)
        SELECT
        	"ct%qe_re""f_2147".col_39576 AS "co) l_39580",
        	regr_count((-5187351000138514782):::INT8::DECIMAL, (-3653536082260194565):::INT8::DECIMAL) OVER (PARTITION BY "ct%qe_re""f_2147".col_39576 ORDER BY "ct%qe_re""f_2147"."co%acl_39579" ASC, "ct%qe_re""f_2147"."%vcol_39578" DESC NULLS FIRST GROUPS 8189510983157791623:::INT8 PRECEDING)::INT8
        		AS col_39581,
        	ARRAY[3159989207:::OID,790553345:::OID,4124327858:::OID,2235099591:::OID,449412673:::OID,2497946448:::OID,3188803936:::OID]
        		AS "c%qo l_39582",
        	'''Vwszd'':1B,31C,182A,428C,648B ''hsRbIsYIb'':559B,665B,902C,926C ''jWCYkhjv'' ''ukNBFB'' ''zMC''':::TSVECTOR
        		AS col_39583
        FROM
        	"w i😿th_7152" AS "ct%qe_re""f_2147"
        GROUP BY
        	"ct%qe_re""f_2147"."c%9bol_39577",
        	"ct%qe_re""f_2147"."co%acl_39579",
        	"ct%qe_re""f_2147"."%vcol_39578",
        	"ct%qe_re""f_2147".col_39576
        HAVING
        	bool_and(true::BOOL)::BOOL
        LIMIT
        	25:::INT8;
```
Schema:

```
    rsg_test.go:718: To reproduce, use schema:
    rsg_test.go:720: SET sql_safe_updates = false;;
    rsg_test.go:720: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:720: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:720: CREATE TABLE table_1 (col1__0 BYTES NOT NULL, "C😏ol1__1" NAME NOT NULL, col1__2 GEOGRAPHY, PRIMARY KEY ("C😏ol1__1" ASC), INDEX (lower("C😏ol1__1")) NOT VISIBLE, UNIQUE (col1__0) PARTITION BY LIST (col1__0) (PARTITION table_1_part_0 VALUES IN (('\x':::BYTES,), ('\x58':::BYTES,), ('\xeac41b94':::BYTES,), ('\xe5':::BYTES,)), PARTITION "ta%6fble_1_part_1" VALUES IN (('\x56424018659a':::BYTES,), ('\x1439b949':::BYTES,), ('\x6d9e40b5945c8fd3':::BYTES,), ('\x3c4806':::BYTES,), ('\x152a3c1d577a9217c3':::BYTES,)), PARTITION "tabl""e_1_part_2" VALUES IN (('\xbfaa':::BYTES,), ('\x03520a456bbe7cca':::BYTES,), ('\xb75a':::BYTES,), ('\x4b77c4126547a9':::BYTES,)), PARTITION taͥble_1_part_3 VALUES IN (('\xd8016b2b2ec00f8dec':::BYTES,), ('\x2c':::BYTES,), ('\x4ec0436b0532f7':::BYTES,), ('\xe17fce1f36':::BYTES,), ('\x55836b2860218b8881':::BYTES,)), PARTITION "ta""ble_1_part_4" VALUES IN (('\x00':::BYTES,), ('\x08e490':::BYTES,), ('\x8822a6':::BYTES,), ('\xecc55500ba05':::BYTES,)), PARTITION "ta%83ble_1_part_5" VALUES IN (('\x9661b6f32ab1675c55':::BYTES,), ('\x27ee8a':::BYTES,), ('\x6e6f479cede3':::BYTES,), ('\x642f1062ec67b4':::BYTES,)), PARTITION table_1_part_6 VALUES IN (('\xf77d':::BYTES,), ('\x1369ba9df0':::BYTES,))) WHERE (table_1.col1__0 = '\x27':::BYTES) OR (table_1."C😏ol1__1" < e'\'':::STRING:::NAME), INVERTED INDEX (col1__0 DESC, "C😏ol1__1", col1__2 ASC), INVERTED INDEX ("C😏ol1__1" ASC, col1__0, col1__2 ASC) VISIBILITY 0.53, UNIQUE ("C😏ol1__1") STORING (col1__0), UNIQUE (col1__0 DESC) PARTITION BY LIST (col1__0) (PARTITION "t{😯able_1_part_0" VALUES IN (('\x8b712fb23de3b4':::BYTES,), ('\xee5179a190ca':::BYTES,), ('\x494f7eb8ef':::BYTES,), ('\x7f':::BYTES,), ('\x':::BYTES,), ('\x3b3d':::BYTES,)), PARTITION table_1_part_1 VALUES IN (('\xab5f5c5cc3198b11':::BYTES,), ('\x003df6022b96':::BYTES,), ('\xab16d6862ff4722207':::BYTES,), ('\xc43e02737b2098':::BYTES,)), PARTITION table_1_part_2 VALUES IN (('\x02a2fb':::BYTES,), ('\xba':::BYTES,), ('\xbea33eac943e21':::BYTES,), ('\x4aac259fa525c5':::BYTES,), ('\xfabf235e1db55f3fd5':::BYTES,), ('\x6f':::BYTES,)), PARTITION table_1_part_3 VALUES IN (('\x1cf5':::BYTES,), ('\xff':::BYTES,), ('\x37f7bcf86e':::BYTES,), ('\xd9e8dd8505':::BYTES,), ('\x24d2':::BYTES,), ('\x0cf1':::BYTES,)), PARTITION table_1_part_4 VALUES IN (('\x4b5666':::BYTES,), ('\x3a2f83495ec64660':::BYTES,), ('\x22':::BYTES,)), PARTITION "tabl e_1_part_5" VALUES IN (('\xa2a9dc784570':::BYTES,), ('\xb35a3b73f2ad9f0089':::BYTES,), ('\x91b415a099b8c04975':::BYTES,)), PARTITION "table_1_part _6" VALUES IN (('\xb987fb':::BYTES,), ('\x9197c5175985d95e10':::BYTES,), ('\x9641c604ad11428a':::BYTES,), ('\x80916ca25a832cfd68':::BYTES,), ('\xea7e83d924d4bbb9':::BYTES,)), PARTITION table_1_part_7 VALUES IN (('\x0c146c':::BYTES,), ('\x4614a78681ab08740f':::BYTES,), ('\x618b38':::BYTES,), ('\x89':::BYTES,)), PARTITION table_1_part_8 VALUES IN (('\x865e846c':::BYTES,), ('\x628644':::BYTES,), ('\x52':::BYTES,), ('\x0ece54':::BYTES,), ('\xd8':::BYTES,))), INVERTED INDEX (col1__2 ASC) VISIBILITY 0.34);
    rsg_test.go:720: CREATE TABLE "tabl e_2" (col2_🙂_0 FLOAT8 NOT NULL, col2__1 STRING NULL, "�col2__2" UUID, col2__3 REFCURSOR NULL, "co\\U000706C4l2__4" FLOAT8 NOT NULL AS (col2_🙂_0 + 0.929467195594817:::FLOAT8) STORED, col2__5 STRING NOT NULL AS (lower(CAST("�col2__2" AS STRING))) STORED, "col'2__6" STRING AS (lower(CAST("�col2__2" AS STRING))) STORED, UNIQUE ("col'2__6", col2__1 DESC) STORING (col2_🙂_0, "�col2__2", "co\\U000706C4l2__4", col2__5), INDEX (col2_🙂_0) WHERE (((("tabl e_2".col2_🙂_0 = (-1.0):::FLOAT8) AND ("tabl e_2"."co\\U000706C4l2__4" <= 1.401298464324817e-45:::FLOAT8)) OR ("tabl e_2".col2__1 >= '"':::STRING)) AND ("tabl e_2".col2__5 <= e'\U00002603':::STRING)) AND ("tabl e_2"."col'2__6" > '':::STRING), INDEX (col2__1 DESC, "�col2__2" ASC, lower(col2__1) ASC), INDEX ("�col2__2" ASC), UNIQUE (col2__5 ASC, lower(CAST("�col2__2" AS STRING)) DESC));
    rsg_test.go:720: CREATE TABLE table_3 (col3__0 OID, "co%pl3__1" INT8 NOT NULL, col3__2 STRING NOT NULL AS (lower(CAST(col3__0 AS STRING))) STORED, UNIQUE ("co%pl3__1" ASC, col3__0 DESC) STORING (col3__2) WHERE (table_3."co%pl3__1" < 32767:::INT8) AND (table_3.col3__2 >= 'X':::STRING), INDEX (lower(CAST(col3__0 AS STRING))) WHERE table_3."co%pl3__1" <= (-1):::INT8, UNIQUE ("co%pl3__1"), INDEX (col3__2, "co%pl3__1" DESC, col3__0 ASC) WHERE table_3."co%pl3__1" > 0:::INT8, INDEX ("co%pl3__1" DESC, col3__0 ASC, col3__2) WHERE (table_3.col3__2 <= e'\x00':::STRING) AND (table_3."co%pl3__1" < 1:::INT8));
    rsg_test.go:720: CREATE TABLE "tablE_4" (col4😸__0 FLOAT4[] NOT NULL, "col4_ 😻_1" TIMESTAMP NOT NULL, col4__2 VARCHAR NOT NULL, col4__3 GEOGRAPHY NULL, "col
        4_,_4" BOX2D, "col4_""_5" TIMESTAMP NOT NULL, col4__6 STRING, col4__7 DATE NOT NULL, col4__8 UUID[] NOT NULL, "C😽ol4__9" "char" NULL, col4__10 INT4 NOT NULL, "co	l4__11" VARCHAR NOT NULL, col4__12 NAME, col4__13 VARCHAR NOT NULL, co̶l4__14 STRING COLLATE da_DK NULL, PRIMARY KEY ("col4_ 😻_1" ASC), UNIQUE ("col4_ 😻_1" DESC) STORING (col4😸__0, col4__2, "col
        4_,_4", col4__6, col4__7, col4__12) WHERE ((((((((("tablE_4"."co	l4__11" > e'\U00002603':::STRING) OR ("tablE_4".col4__13 < e'\U00002603':::STRING)) AND ("tablE_4"."C😽ol4__9" > 'X':::STRING)) AND ("tablE_4".col4__6 != e'\x00':::STRING)) OR ("tablE_4"."col4_ 😻_1" > '0001-01-01 00:00:00':::TIMESTAMP)) AND ("tablE_4".col4__10 < 1:::INT8)) AND ("tablE_4".col4__12 >= e'\x00':::STRING:::NAME)) OR ("tablE_4".col4__7 > '5874897-12-31':::DATE)) AND ("tablE_4".col4__2 > e'\'':::STRING)) AND ("tablE_4"."col4_""_5" != '0001-01-01 00:00:00':::TIMESTAMP), INDEX (col4__8 ASC) STORING (col4__2, "col4_""_5", col4__6, "C😽ol4__9", col4__10) WHERE (((((((("tablE_4".col4__2 < e'\U00002603':::STRING) OR ("tablE_4"."col4_ 😻_1" > '4714-11-24 00:00:00 BC':::TIMESTAMP)) AND ("tablE_4".col4__10 < (-2147483648):::INT8)) AND ("tablE_4"."C😽ol4__9" > '"':::STRING)) OR ("tablE_4"."co	l4__11" >= e'\U00002603':::STRING)) AND ("tablE_4".col4__7 < '-infinity':::DATE)) OR ("tablE_4"."col4_""_5" >= '294276-12-31 23:59:59.999999':::TIMESTAMP)) OR ("tablE_4".col4__6 < '':::STRING)) AND ("tablE_4".col4__12 < 'X':::STRING:::NAME) NOT VISIBLE, FAMILY ("co	l4__11", "col4_""_5", col4__3, col4__10), FAMILY ("col
        4_,_4"), FAMILY (col4__8), FAMILY (col4__13, "C😽ol4__9"), FAMILY (col4__2, col4__7, col4__12, col4😸__0), FAMILY (col4__6), FAMILY ("col4_ 😻_1", co̶l4__14));
    rsg_test.go:720: CREATE TABLE "�t😶able_5" ("co%ql5__0" PG_LSN NOT NULL, col5__1 INTERVAL NOT NULL, col5__2 FLOAT8, col😋5_😱_3 TIME, col😛5__4 TSQUERY NOT NULL, col5__5 FLOAT8, "co😲l5_%p_6" TSQUERY NOT NULL, "c'o l5ͨ__7" PG_LSN NOT NULL, col5__8 BOOL NOT NULL, col5__9 OID, "col5_""_10" UUID NOT NULL, col5__11 BOOL NOT NULL, col5__12 REGCLASS, "col5_�_13" TSQUERY NULL, col5__14 FLOAT8 AS (col5__2 + col5__5) VIRTUAL, " Col5__15" STRING NOT NULL AS (lower(CAST("col5_""_10" AS STRING))) STORED, col5__16 STRING NOT NULL AS (lower(CAST(col5__8 AS STRING))) STORED, col5__17 STRING AS (lower(CAST(col😋5_😱_3 AS STRING))) STORED, FAMILY ("co%ql5__0", "co😲l5_%p_6"), FAMILY ("col5_�_13", col😋5_😱_3), FAMILY ("col5_""_10"), FAMILY (col5__16, col5__2, col5__17), FAMILY (col😛5__4, "c'o l5ͨ__7", col5__5), FAMILY (col5__9, col5__12, " Col5__15"), FAMILY (col5__11), FAMILY (col5__8, col5__1));
    rsg_test.go:720: ALTER TABLE table_1 INJECT STATISTICS e'[{"avg_size": 23, "columns": ["col1__0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 48, "histo_col_type": "BYTES", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 70}, {"avg_size": 24, "columns": ["\\"C\U0001F60Fol1__1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 43, "histo_buckets": [{"distinct_range": 0, "num_eq": 6262050362911140380, "num_range": 0, "upper_bound": "\\u001c\\u0011"}, {"distinct_range": 0, "num_eq": 0, "num_range": 4000, "upper_bound": "*o"}, {"distinct_range": 0, "num_eq": 5115606792080235203, "num_range": 500000, "upper_bound": "+v\\u0010X8]"}, {"distinct_range": 8828098520937086000, "num_eq": 1672446796283186675, "num_range": 8828098520937086271, "upper_bound": "V\\u001dKR=\\u0007w"}, {"distinct_range": 6191793832.963597, "num_eq": 245854498584241124, "num_range": 10000000000, "upper_bound": "pz!Tn\\u0013Q"}], "histo_col_type": "NAME", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 70}, {"avg_size": 3, "columns": ["col1__2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 23, "histo_buckets": [{"distinct_range": 0, "num_eq": 1226533234025365765, "num_range": 0, "upper_bound": "\\\\x42fd100000000000000000bff4a7b2f25c2273bff91e2cf71f73c14007048a702575743ff81fc55eab06d8"}, {"distinct_range": 1730798349156838700, "num_eq": 5000000, "num_range": 1730798349156838751, "upper_bound": "\\\\x42fd100000000000000000c00921fb54442d18bff90769ff1625fb400921fb54442d183ff8d5adc524ee85"}, {"distinct_range": 5000000, "num_eq": 5961757391420845858, "num_range": 5000000, "upper_bound": "\\\\x42fd300000000000000000bff4a7b2f25c2273bff91e2cf71f73c14007048a702575743ff81fc55eab06d8"}, {"distinct_range": 0, "num_eq": 1601839503258133256, "num_range": 40000000000, "upper_bound": "\\\\x42fd300000000000000000c00921fb54442d18bff90769ff1625fb400921fb54442d183ff8d5adc524ee85"}, {"distinct_range": 0, "num_eq": 1260018055195123352, "num_range": 4771121217132924665, "upper_bound": "\\\\x42fd500000000000000000bff4a7b2f25c2273bff91e2cf71f73c14007048a702575743ff81fc55eab06d8"}, {"distinct_range": 0, "num_eq": 5667992140551896456, "num_range": 500, "upper_bound": "\\\\x42fd500000000000000000c005aae054b37ef0bff832bdb59054a040087e9a4db86d6e3ff8ce9bfd93ba88"}, {"distinct_range": 274431.90898135194, "num_eq": 6707254466684227990, "num_range": 300000, "upper_bound": "\\\\x42fd500000000000000000c00921fb54442d18bff90769ff1625fb400921fb54442d183ff8d5adc524ee85"}, {"distinct_range": 3929509775968871000, "num_eq": 8734115789468585894, "num_range": 3929509775968870831, "upper_bound": "\\\\x42fd640000000000000000c005aae054b37ef0bff832bdb59054a040087e9a4db86d6e3ff8ce9bfd93ba88"}, {"distinct_range": 100, "num_eq": 5510641792289350568, "num_range": 100, "upper_bound": "\\\\x42fd6c0000000000000000c005aae054b37ef0bff832bdb59054a040087e9a4db86d6e3ff8ce9bfd93ba88"}, {"distinct_range": 0, "num_eq": 100000000000, "num_range": 0, "upper_bound": "\\\\x42fd700000000000000000c00921fb54442d18bff90769ff1625fb400921fb54442d183ff8d5adc524ee85"}, {"distinct_range": 400000, "num_eq": 800000, "num_range": 400000, "upper_bound": "\\\\x42fd8c0000000000000000bff4a7b2f25c2273bff91e2cf71f73c14007048a702575743ff81fc55eab06d8"}, {"distinct_range": 4372289475542780400, "num_eq": 100000, "num_range": 5114848097451961914, "upper_bound": "\\\\x42fd900000000000000000c00921fb54442d18bff90769ff1625fb400921fb54442d183ff8d5adc524ee85"}, {"distinct_range": 1285878.4320782223, "num_eq": 6000000000, "num_range": 2000000, "upper_bound": "\\\\x42fdb00000000000000000bff4a7b2f25c2273bff91e2cf71f73c14007048a702575743ff81fc55eab06d8"}, {"distinct_range": 0, "num_eq": 1882488892573353518, "num_range": 5015193261379190888, "upper_bound": "\\\\x42fdb00000000000000000c005aae054b37ef0bff832bdb59054a040087e9a4db86d6e3ff8ce9bfd93ba88"}, {"distinct_range": 0, "num_eq": 3985551795293794645, "num_range": 1001531740509294846, "upper_bound": "\\\\x42fdb00000000000000000c00921fb54442d18bff90769ff1625fb400921fb54442d183ff8d5adc524ee85"}], "histo_col_type": "BYTES", "histo_version": 3, "name": "__auto__", "null_count": 46, "row_count": 70}]':::JSONB;
    rsg_test.go:720: ALTER TABLE "tabl e_2" INJECT STATISTICS e'[{"avg_size": 22, "columns": ["col2_\U0001F642_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 13369113, "histo_buckets": [{"distinct_range": 0, "num_eq": 100000000000, "num_range": 0, "upper_bound": "0.3334753099412011"}, {"distinct_range": 2035190830668209200, "num_eq": 800000, "num_range": 4400082989317337560, "upper_bound": "0.8682465942258499"}, {"distinct_range": 80000000, "num_eq": 3000000000, "num_range": 80000000, "upper_bound": "1.254893184653974"}, {"distinct_range": 0, "num_eq": 8412410977047688139, "num_range": 10000000, "upper_bound": "1.6305287975172313"}, {"distinct_range": 38044488.75300038, "num_eq": 1000000, "num_range": 40000000, "upper_bound": "-0.5304737752826678"}], "histo_col_type": "FLOAT8", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 400000000}, {"avg_size": 22, "columns": ["col2__1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 269204527, "histo_buckets": [{"distinct_range": 0, "num_eq": 5259436989830042462, "num_range": 0, "upper_bound": ""}, {"distinct_range": 0, "num_eq": 5487665993591238941, "num_range": 1000000, "upper_bound": "L\\u001eG"}, {"distinct_range": 2020839645511210200, "num_eq": 3561926238805134033, "num_range": 5524752032544166847, "upper_bound": "0x~\\u0013"}, {"distinct_range": 900, "num_eq": 5374840956102249681, "num_range": 900, "upper_bound": "Y\\u0015J]u*"}, {"distinct_range": 7275176848662196000, "num_eq": 3000000, "num_range": 7275176848662196648, "upper_bound": "1\\u001eo88\\u0003E"}, {"distinct_range": 0, "num_eq": 5000000, "num_range": 212948754670289244, "upper_bound": "s>p{3A5y"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 310615881, "row_count": 400000000}, {"avg_size": 20, "columns": ["\\"\\u000bcol2__2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 387455554, "histo_buckets": [{"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": "051b8047-4308-46b2-8c54-598118523b04"}, {"distinct_range": 100000000000, "num_eq": 823547170962949782, "num_range": 100000000000, "upper_bound": "5698597a-cba6-462c-8b40-9a1aeca005cb"}, {"distinct_range": 9000, "num_eq": 30, "num_range": 9000, "upper_bound": "6b03cd7e-7235-4401-bf5b-7fe94900a310"}, {"distinct_range": 200000000, "num_eq": 3646380858923422738, "num_range": 200000000, "upper_bound": "94de0b73-bc1a-4156-82dc-f60b6f1aee1a"}, {"distinct_range": 70000000000, "num_eq": 6000000, "num_range": 70000000000, "upper_bound": "d3620387-ba04-4233-b237-ade2da55270f"}, {"distinct_range": 43.67114548490963, "num_eq": 3118826037844278399, "num_range": 500, "upper_bound": "f688dba4-496a-4ac0-8050-f241c38b40ad"}], "histo_col_type": "UUID", "histo_version": 3, "name": "__auto__", "null_count": 171696441, "row_count": 400000000}, {"avg_size": 6, "columns": ["col2__3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 104440505, "histo_col_type": "", "name": "__auto__", "null_count": 190261748, "row_count": 400000000}, {"avg_size": 4, "columns": ["\\"co\\\\\\\\U000706C4l2__4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6625630, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 400000000}, {"avg_size": 2, "columns": ["col2__5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 293188084, "histo_buckets": [{"distinct_range": 0, "num_eq": 4957896519801550828, "num_range": 0, "upper_bound": ""}, {"distinct_range": 0, "num_eq": 2133266837737591622, "num_range": 400000, "upper_bound": "A"}, {"distinct_range": 0, "num_eq": 400000, "num_range": 0, "upper_bound": "\\u000fu"}, {"distinct_range": 7174789422945303000, "num_eq": 1000000, "num_range": 7174789422945302965, "upper_bound": "e^\\u0017\\\\"}, {"distinct_range": 40000, "num_eq": 6536136426735781064, "num_range": 40000, "upper_bound": "Wg``Cf"}, {"distinct_range": 0, "num_eq": 1560328924711034893, "num_range": 900000, "upper_bound": "P\\u0006Sz0s\\u000e"}, {"distinct_range": 26187647.890027676, "num_eq": 6494341991094986935, "num_range": 40000000, "upper_bound": "``u^\\u001ch\\u000f6\'\'"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 400000000}, {"avg_size": 17, "columns": ["\\"col\'2__6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 232989071, "histo_buckets": [{"distinct_range": 0, "num_eq": 4271291149443303997, "num_range": 0, "upper_bound": ""}, {"distinct_range": 100000000, "num_eq": 1725597908694742389, "num_range": 100000000, "upper_bound": "\\n"}, {"distinct_range": 389939271239488060, "num_eq": 4736550629202316604, "num_range": 1760579026802957698, "upper_bound": "33"}, {"distinct_range": 3393458.688146998, "num_eq": 2000, "num_range": 20000000, "upper_bound": "{\\u000cf"}, {"distinct_range": 742848926261343900, "num_eq": 20000, "num_range": 4048497120265225455, "upper_bound": "Nz@N"}, {"distinct_range": 0, "num_eq": 3494174629369066346, "num_range": 6233106069623472181, "upper_bound": "OF\\u0005>``"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 350567009, "row_count": 400000000}]':::JSONB;
    rsg_test.go:720: ALTER TABLE table_3 INJECT STATISTICS e'[{"avg_size": 15, "columns": ["col3__0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 27851, "histo_col_type": "", "name": "__auto__", "null_count": 14510, "row_count": 40000}, {"avg_size": 16, "columns": ["\\"co%pl3__1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 26587, "histo_buckets": [{"distinct_range": 0, "num_eq": 6208730453515277036, "num_range": 0, "upper_bound": "2817919109093821465"}, {"distinct_range": 30000000000, "num_eq": 40000000, "num_range": 30000000000, "upper_bound": "4409183533453716826"}, {"distinct_range": 2798747713615449600, "num_eq": 1722182934408246387, "num_range": 8340350706819483495, "upper_bound": "-8559901335052269098"}, {"distinct_range": 1891775286943126000, "num_eq": 789511353679052744, "num_range": 2746703279499665983, "upper_bound": "2502863737738329518"}, {"distinct_range": 2121026528246738200, "num_eq": 10000000, "num_range": 2121026528246738071, "upper_bound": "-672221643483938804"}, {"distinct_range": 0, "num_eq": 7014088095611243212, "num_range": 10000, "upper_bound": "8774832774107571837"}, {"distinct_range": 30000000, "num_eq": 8801778717252548035, "num_range": 30000000, "upper_bound": "-8136558729851804608"}, {"distinct_range": 40000000, "num_eq": 6000000000, "num_range": 40000000, "upper_bound": "-32768"}], "histo_col_type": "INT8", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 40000}, {"avg_size": 29, "columns": ["col3__2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 206, "histo_buckets": [{"distinct_range": 0, "num_eq": 10000000, "num_range": 0, "upper_bound": "#"}, {"distinct_range": 830114236085536600, "num_eq": 10000000, "num_range": 830114236085536655, "upper_bound": "(x\\u001a"}, {"distinct_range": 90000000, "num_eq": 400, "num_range": 90000000, "upper_bound": "=\\u0011!"}, {"distinct_range": 0, "num_eq": 6000000000, "num_range": 10000000000, "upper_bound": ")kw~"}, {"distinct_range": 0, "num_eq": 8988532491264512261, "num_range": 1776152339172839825, "upper_bound": "6\\n=\\u00085"}, {"distinct_range": 90, "num_eq": 2657699332151209503, "num_range": 90, "upper_bound": "Y\\u0017$:*"}, {"distinct_range": 0, "num_eq": 5464867630780964725, "num_range": 6411703911090703316, "upper_bound": "yB4 x"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 40000}]':::JSONB;
    rsg_test.go:720: ALTER TABLE "tablE_4" INJECT STATISTICS e'[{"avg_size": 14, "columns": ["col4__2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 74823216251, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 80000000000}, {"avg_size": 6, "columns": ["col4__6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 16704103836, "histo_col_type": "", "name": "__auto__", "null_count": 32483078261, "row_count": 80000000000}, {"avg_size": 26, "columns": ["\\"C\U0001F63Dol4__9\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 27680898603, "histo_col_type": "", "name": "__auto__", "null_count": 53608423303, "row_count": 80000000000}, {"avg_size": 6, "columns": ["co\u0336l4__14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 25418553043, "histo_col_type": "", "name": "__auto__", "null_count": 44612922637, "row_count": 80000000000}, {"avg_size": 6, "columns": ["\\"col4_ \U0001F63B_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 62486318488, "histo_buckets": [{"distinct_range": 0, "num_eq": 100, "num_range": 0, "upper_bound": "2004-03-23 22:14:24.00087"}, {"distinct_range": 8439541844939435000, "num_eq": 7392071223652113477, "num_range": 8439541844939434631, "upper_bound": "2032-05-12 06:15:36.000338"}, {"distinct_range": 70000000000, "num_eq": 200000, "num_range": 70000000000, "upper_bound": "1980-11-14 04:06:00.000984"}, {"distinct_range": 7.930845590288191, "num_eq": 4138439709305537038, "num_range": 10, "upper_bound": "2032-03-28 03:10:35.000867"}, {"distinct_range": 0, "num_eq": 6000, "num_range": 80000000000, "upper_bound": "2000-11-16 05:52:50.000464"}, {"distinct_range": 0, "num_eq": 7000000, "num_range": 7250691120678166157, "upper_bound": "1982-03-30 20:48:57.000752"}, {"distinct_range": 3000000, "num_eq": 3265490930275000721, "num_range": 3000000, "upper_bound": "294276-12-31 23:59:59.999999"}], "histo_col_type": "TIMESTAMP", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 80000000000}, {"avg_size": 2, "columns": ["col4__8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 32613210671, "histo_buckets": [{"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": "ARRAY[]"}, {"distinct_range": 0, "num_eq": 400000, "num_range": 1482267037774648320, "upper_bound": "ARRAY[\'1672fa81-8c1f-42ff-a6d5-cb47797597b5\':::UUID,\'2541a506-18c2-4dc6-935b-fbe9f23e76c6\':::UUID,\'3266eaae-b71b-46e2-ba5c-617a8af36194\':::UUID,\'835d3f20-2674-4478-a6e5-ea5ad64f61ee\':::UUID]"}, {"distinct_range": 674569872420312800, "num_eq": 400000, "num_range": 2822095457079858954, "upper_bound": "ARRAY[\'dc95757d-fde4-465e-80d4-57df8279f17f\':::UUID,\'483e926c-d37a-43c1-b516-4a07c473c2e7\':::UUID,\'9e1923ab-6ea6-4324-a415-961ce937eb42\':::UUID,\'bf137599-99f9-4c9f-bc7e-664991b7653f\':::UUID]"}], "histo_col_type": "UUID[]", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 80000000000}, {"avg_size": 1, "columns": ["col4__3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 43970897548, "histo_col_type": "", "name": "__auto__", "null_count": 68976151912, "row_count": 80000000000}, {"avg_size": 25, "columns": ["\\"col\\n4_,_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 30657845211, "histo_col_type": "", "name": "__auto__", "null_count": 73602789441, "row_count": 80000000000}, {"avg_size": 26, "columns": ["col4__7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 69902751869, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 80000000000}, {"avg_size": 1, "columns": ["col4__10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 76648349969, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 80000000000}, {"avg_size": 18, "columns": ["\\"co\\tl4__11\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 73855245890, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 80000000000}, {"avg_size": 24, "columns": ["col4__12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 75245332839, "histo_col_type": "", "name": "__auto__", "null_count": 10634060071, "row_count": 80000000000}, {"avg_size": 10, "columns": ["col4__13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 32752256362, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 80000000000}, {"avg_size": 16, "columns": ["col4\U0001F638__0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 51179030913, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 80000000000}, {"avg_size": 19, "columns": ["\\"col4_\\"\\"_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 42514422729, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 80000000000}]':::JSONB;
    rsg_test.go:720: ALTER TABLE "�t😶able_5" INJECT STATISTICS e'[{"avg_size": 0, "columns": ["col\U0001F60B5_\U0001F631_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3569611139582680385, "histo_col_type": "", "name": "__auto__", "null_count": 3183274939406093233, "row_count": 6818160101660957652}, {"avg_size": 6, "columns": ["col5__5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4658976942867647902, "histo_col_type": "", "name": "__auto__", "null_count": 3801941775416016261, "row_count": 6818160101660957652}, {"avg_size": 0, "columns": ["col5__12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5277000309090317688, "histo_col_type": "", "name": "__auto__", "null_count": 5295937900675559185, "row_count": 6818160101660957652}, {"avg_size": 6, "columns": ["col5__17"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5494744618684239982, "histo_col_type": "", "name": "__auto__", "null_count": 6314263386223219591, "row_count": 6818160101660957652}, {"avg_size": 19, "columns": ["col5__2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 860308777031690663, "histo_col_type": "", "name": "__auto__", "null_count": 3518712325737746943, "row_count": 6818160101660957652}, {"avg_size": 0, "columns": ["col5__9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2107159671209841592, "histo_col_type": "", "name": "__auto__", "null_count": 3857237865896350193, "row_count": 6818160101660957652}, {"avg_size": 11, "columns": ["col5__16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 533054294865212725, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}, {"avg_size": 29, "columns": ["\\"co%ql5__0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4214706067261457785, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}, {"avg_size": 13, "columns": ["\\"col5_\\"\\"_10\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5976604447856876960, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}, {"avg_size": 19, "columns": ["\\" Col5__15\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 246197636049216475, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}, {"avg_size": 28, "columns": ["col\U0001F61B5__4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3651306916055901754, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}, {"avg_size": 17, "columns": ["\\"co\U0001F632l5_%p_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 364447955383867803, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}, {"avg_size": 23, "columns": ["\\"c\'o l5\u0368__7\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1664890430908323466, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}, {"avg_size": 2, "columns": ["col5__8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5484385579083749283, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}, {"avg_size": 14, "columns": ["col5__11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1550284269831502130, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}, {"avg_size": 18, "columns": ["\\"col5_\\u000b_13\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 906479642836607741, "histo_col_type": "", "name": "__auto__", "null_count": 5078044732340743594, "row_count": 6818160101660957652}, {"avg_size": 22, "columns": ["col5__14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4690276292172289378, "histo_col_type": "", "name": "__auto__", "null_count": 5147263469285302170, "row_count": 6818160101660957652}, {"avg_size": 15, "columns": ["col5__1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2698338691945477206, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6818160101660957652}]':::JSONB;
    rsg_test.go:720: CREATE TYPE rand_typ_0 AS ENUM ('ffl', 'ecd', 'axqaya', 'ztuc', 'eq', 'wc');
    rsg_test.go:720: CREATE TYPE rand_typ_1 AS ENUM ('wbqv', 'y', 'fspufx', 'my');
    rsg_test.go:720: SET sql_safe_updates = false;;
    rsg_test.go:720: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (8045733038883859860);
    rsg_test.go:720: ALTER TABLE seed SCATTER;
    rsg_test.go:722: 
    rsg_test.go:723: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith2520457243
--- FAIL: TestRandomSyntaxSQLSmith (376.66s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13830529?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13830529?buildTab=artifacts#/) on master @ [3c41c509a87cba7a1fd3f5cfdb0f6badb78e3704](https://github.com/cockroachdb/cockroach/commits/3c41c509a87cba7a1fd3f5cfdb0f6badb78e3704):

Random syntax error:

```
    rsg_test.go:909: Crash detected: server panic: pq: internal error: unable to encode table key: *tree.DTSVector
```
Query:

```
        WITH
        	"wi\\xc4tH_71730" (co😣l_426908)
        		AS (
        			SELECT
        				*
        			FROM
        				(VALUES (1057278813:::OID), (NULL), (3037283361:::OID), (0:::OID)) AS "%pta&b_175654" (co😣l_426908)
        		),
        	"Wi̲th%p_71733" (col_426912, col_426913)
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						(
        							'106.221.125.203/13':::INET,
        							('''AayKgrgJQI'' ''Hd'' ''MgtRusLNK'':600B,964C ''REsv'':411A,853C ''Wej'' ''lGrgCCt'':53A,281A,656C,826C,844B ''zmt'':296C,959B':::TSVECTOR, 'FFFFFFFF/FFFFFFFF':::PG_LSN)
        						),
        						(
        							('5c6:6525:dc81:c358:8abe:a699:b629:e2dc/117':::INET::INET | '4.61.186.45/1':::INET::INET)::INET,
        							('''PksX'':51B,530B,762C,929C ''hiHb'':559B,657B,709C,907C,940C ''oVYMvUwA'':75A,300B,651B,794C ''wKQp''':::TSVECTOR, 'B3DA038B/37756B67':::PG_LSN)
        						),
        						(NULL, ('''dYNPcaZsq'':220B,710B,991C':::TSVECTOR, 'C7887B6E/B4D971A4':::PG_LSN)),
        						(
        							('95c4:31d:bda0:7759:3633:5a5f:6dc4:4d32/69':::INET::INET & inet('14fa:3f6d:462a:b3f3:5d67:e290:218c:9840/3':::INET::INET)::INET::INET)::INET,
        							('''F'' ''IHfJxzRKHW'':51A ''NChB'' ''XScS'' ''ggNWTS'' ''sCB'' ''yrdSiOp'':679A,778B,833B':::TSVECTOR, 'E43F9A1C/5E3CF0FB':::PG_LSN)
        						),
        						(
        							'74b1:91:d3d0:7ae0:c4dd:3e70:cfbe:87af/14':::INET,
        							(
        								WITH
        									with_71731 (col_426909)
        										AS (
        											SELECT
        												*
        											FROM
        												(
        													VALUES
        														(e'\U0004730D\U000DBF75\U000D162D\U000BAE25' COLLATE en_US),
        														(e'\U00041ECA\U0005F9F9\U00017EE7' COLLATE en_US),
        														(e'\U000EBC62\U0001C3CD\U0008FCBE' COLLATE en_US),
        														(
        															e'\U000AC22B\U00001067\U0004AB47\U000E3EFB\U00076076\U000B9352\U000322FA\U0002A8DF\U00064B5C' COLLATE en_US
        														)
        												)
        													AS tab_175655 (col_426909)
        										),
        									with_71732 ("\vcol_426910")
        										AS (
        											SELECT
        												*
        											FROM
        												(VALUES (NULL), (NULL), ('':::VOID)) AS t͉ab_175656 ("\vcol_426910")
        										)
        								SELECT
        									('''bKnLfKuR'' ''gslDKhpD'':94A,761B ''qYDm''':::TSVECTOR, 'F806BF3F/77F47A98':::PG_LSN)
        										AS col_426911
        								FROM
        									defaultdb.public.table_1@[0] AS "\\U00082ADDta�b_175657"
        								LIMIT
        									1:::INT8
        							)
        						)
        				)
        					AS tab_175658 (col_426912, col_426913)
        		)
        SELECT
        	cte_ref_20796.col_426913 AS "C ol_426914"
        FROM
        	"Wi̲th%p_71733" AS cte_ref_20796
        GROUP BY
        	cte_ref_20796.col_426913
        LIMIT
        	30:::INT8;
```
Schema:

```
    rsg_test.go:718: To reproduce, use schema:
    rsg_test.go:720: SET sql_safe_updates = false;;
    rsg_test.go:720: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:720: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:720: CREATE TABLE table_1 (col1__0 INTERVAL NOT NULL, "col1(__1" BYTES NOT NULL, "co%pl1__2" REFCURSOR NOT NULL, "c%46ol1__3" GEOMETRY NULL, "co-ḻ1.__4" TIME NOT NULL, "c/ol1__5" GEOMETRY, "Col😠1__6" INET NOT NULL, col😄1__7 "char" NOT NULL, col1__8 REGROLE NULL, "col1 
__9" INT8, col1__10 STRING NULL AS (CASE WHEN col1__8 IS NULL THEN e'LcDp[9\x13k':::STRING ELSE e'a;(>HkD\x16z':::STRING END) STORED, col1__11 STRING NOT NULL AS (lower(CAST("Col😠1__6" AS STRING))) STORED, "cOl1__12" STRING AS (lower(CAST("c/ol1__5" AS STRING))) STORED);
    rsg_test.go:720: ALTER TABLE table_1 INJECT STATISTICS e'[{"avg_size": 20, "columns": ["\\"c/ol1__5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2866696327, "histo_col_type": "", "name": "__auto__", "null_count": 1862021971, "row_count": 6000000000}, {"avg_size": 25, "columns": ["col\U0001F6041__7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 9521464, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6000000000}, {"avg_size": 3, "columns": ["\\"col1 \\r__9\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1014220736, "histo_col_type": "", "name": "__auto__", "null_count": 4676886763, "row_count": 6000000000}, {"avg_size": 15, "columns": ["\\"co%pl1__2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3290248247, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6000000000}, {"avg_size": 1, "columns": ["col1__8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5191035716, "histo_col_type": "", "name": "__auto__", "null_count": 5247463480, "row_count": 6000000000}, {"avg_size": 4, "columns": ["col1__11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2399981025, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6000000000}, {"avg_size": 12, "columns": ["col1__0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1038853859, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6000000000}, {"avg_size": 8, "columns": ["\\"col1(__1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2976344396, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6000000000}, {"avg_size": 21, "columns": ["\\"c%46ol1__3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1336080485, "histo_col_type": "", "name": "__auto__", "null_count": 495462538, "row_count": 6000000000}, {"avg_size": 6, "columns": ["col1__10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5605429831, "histo_col_type": "", "name": "__auto__", "null_count": 2416434619, "row_count": 6000000000}, {"avg_size": 19, "columns": ["\\"co-\U00001E3B1.__4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4983805628, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6000000000}, {"avg_size": 22, "columns": ["\\"Col\U0001F6201__6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2858107216, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6000000000}, {"avg_size": 1, "columns": ["\\"cOl1__12\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 775431886, "histo_col_type": "", "name": "__auto__", "null_count": 3314105552, "row_count": 6000000000}]':::JSONB;
    rsg_test.go:720: CREATE TYPE rand_typ_0 AS ENUM ('gwc', 'zfbyu', 'toowp');
    rsg_test.go:720: SET sql_safe_updates = false;;
    rsg_test.go:720: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (2786580106065635261);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (5370492709792488660);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (35454307210594801);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (4511197293181160997);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (7409219263946762723);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (5873114608656185759);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (8742904838702090582);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (8566859963350822110);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (2178501654566791712);
    rsg_test.go:720: ALTER TABLE seed SCATTER;
    rsg_test.go:722: 
    rsg_test.go:723: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith3157615624
--- FAIL: TestRandomSyntaxSQLSmith (385.63s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13840938?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13840938?buildTab=artifacts#/) on master @ [d272e9ef5589deff570efc023db6c70edfde311c](https://github.com/cockroachdb/cockroach/commits/d272e9ef5589deff570efc023db6c70edfde311c):

Random syntax error:

```
    rsg_test.go:909: Crash detected: server panic: pq: internal error: unable to encode table key: *tree.DTSQuery
```
Query:

```
        WITH
        	"%vwith_36164" ("c""ol_202416")
        		AS (
        			SELECT
        				B'000110100100010110100111001101011100010101100' AS "c""ol_202416"
        			FROM
        				defaultdb.public."ta%qble_2"@"ta%qble_2_col2_""_1_expr_key" AS tab_88011,
        				defaultdb.public.table_1@[0] AS ta̜b_88012
        				JOIN defaultdb.public."ta%qble_2"@"ta%qble_2_col2_""_1_expr_key" AS tab_88013 ON
        						(ta̜b_88012."%vcol1_	_2") = (tab_88013.col2__0)
        		),
        	w̆ith_36165 ("c%pol_202417", col_202418)
        		AS (
        			SELECT
        				NULL AS "c%pol_202417", CASE WHEN false THEN NULL ELSE NULL END AS col_202418
        			FROM
        				"%vwith_36164" AS "cte_reͬf%6d_10486"
        			WHERE
        				true
        			INTERSECT ALL
        				SELECT
        					*
        				FROM
        					(
        						VALUES
        							(
        								(e'9#9~^\x05D':::STRING, '''dD'' | !''iTFgYS'' | ''Fz'' | ''eYYBGMODE'' <-> ''gHEhmeTfFL''':::TSQUERY),
        								'pihyqc':::rand_typ_4
        							),
        							((e'PM\x11A]\'':::STRING, '!''n'' & ''z'' | ''jcugzJNnf''':::TSQUERY), NULL),
        							(
        								(e'1Z``$\fD6':::STRING, '''XfbkUMtNKs'' & ''QMxyEgb'' | ''Dnlb''':::TSQUERY),
        								'pihyqc':::rand_typ_4
        							),
        							(('':::STRING, '''TZA'' | ''zUgwhmgiw''':::TSQUERY), NULL)
        					)
        						AS tab_88014 (col_202419, col_202420)
        		),
        	with_36166 (col_202421)
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						('''XCDHlEhzup'' ''ikTKEO'':371A,542A,929B':::TSVECTOR),
        						(
        							'''H'':59B,171C,656B,683A ''KMjfuqImav'':548C,720B ''MiTdpPuFA'':88C,767A,929C ''nHzMu''':::TSVECTOR
        						),
        						(
        							'''XSdvE'':485B,731A,914A ''ZmU'' ''dGjxsleHg'' ''iNfj'':188A,557A,608B,676A ''jqftayN''':::TSVECTOR
        						),
        						(
        							'''Ce'':208B,445A ''DTRUM'':175A,390C,536B,758A ''Hbu'':84A,99B,539A,906C ''JCekHM'' ''SbsGG'':487B,711C ''SfMKGb'':78B,625B,652B ''UQXzRWZAtr'' ''fBi'' ''oP'' ''wA''':::TSVECTOR
        						),
        						(
        							'''DGFV'' ''DudTGY'':400C,908C ''EECPNnpUVQ'':943A ''EZ'' ''MmyTXhegU'':544C,595C,683B ''QvgefwW'':359B ''mkyG'':31A,466A ''sHVwJGgCmR'':113B,470A ''tElksU'':385B,591A,676A,979A':::TSVECTOR
        						),
        						(
        							'''HWYPVt'':254B,282A,430B,464A ''MODiUTjS'':477B,857B ''mSjAN'' ''maxfzsjiUi'':62B,999B':::TSVECTOR
        						)
        				)
        					AS "taB_88015" (col_202421)
        		)
        SELECT
        	CASE
        	WHEN true THEN cte_ref_10487."c%pol_202417"
        	ELSE (e'!\x04%\x1eH':::STRING, '''xQhOioXFAX'' & ''CH''':::TSQUERY)
        	END
        		AS col_202422,
        	'pihyqc':::rand_typ_4 AS col_202423
        FROM
        	w̆ith_36165 AS cte_ref_10487
        WHERE
        	true
        ORDER BY
        	cte_ref_10487.col_202418 DESC NULLS LAST, cte_ref_10487."c%pol_202417", cte_ref_10487.col_202418 ASC
        LIMIT
        	78:::INT8;
```
Schema:

```
    rsg_test.go:718: To reproduce, use schema:
    rsg_test.go:720: SET sql_safe_updates = false;;
    rsg_test.go:720: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:720: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:720: CREATE TABLE table_1 ("co l1__0" TSVECTOR NOT NULL, "col1_""_1" STRING, "%vcol1_	_2" INT2 NOT NULL, col1__3 STRING NOT NULL AS (lower("col1_""_1")) STORED);
    rsg_test.go:720: CREATE TABLE "ta%qble_2" (col2__0 INT4 NOT NULL, "col2_""_1" VARCHAR NOT NULL, "c?ol2__2" TIMESTAMPTZ NOT NULL, "c o%q😹l2__3" STRING COLLATE da_DK NOT NULL, col2__4 BYTES NOT NULL, "col2_�_5" TIMETZ NOT NULL, "col%q2__6" REGCLASS NULL, col2__7 FLOAT8 NOT NULL, col2__8 TSQUERY NOT NULL, "co\vl2_%p_9" FLOAT8 NOT NULL AS (col2__7 + 0.8271012741471613:::FLOAT8) VIRTUAL, col2__10 STRING NOT NULL AS (lower(CAST("col2_�_5" AS STRING))) VIRTUAL, col😆2__11 INT8 NOT NULL AS (col2__0 + 1746478085:::INT8) STORED, col2__12 STRING NOT NULL AS (lower(CAST("col2_�_5" AS STRING))) VIRTUAL, col2__13 STRING NOT NULL AS (CASE WHEN col2__4 IS NULL THEN 'V#q54}':::STRING ELSE e'-\x1fml[&*':::STRING END) STORED, UNIQUE (col2__4) STORING (col2__0, "col2_""_1", "c?ol2__2", "c o%q😹l2__3", "col2_�_5", "col%q2__6", col😆2__11), UNIQUE (col2__4 DESC) STORING (col2__0, "col2_""_1", "c?ol2__2", col😆2__11, col2__13), UNIQUE ("co\vl2_%p_9") STORING ("col2_""_1", "c?ol2__2", "c o%q😹l2__3", col2__4, "col2_�_5", "col%q2__6") WHERE ((((((((("ta%qble_2".col2__12 = 'X':::STRING) OR ("ta%qble_2"."col2_""_1" >= e'\x00':::STRING)) OR ("ta%qble_2".col2__13 < e'\U00002603':::STRING)) AND ("ta%qble_2"."col2_�_5" = '24:00:00-15:59':::TIMETZ)) OR ("ta%qble_2"."c?ol2__2" > '3000-01-01 00:00:00+00':::TIMESTAMPTZ)) AND ("ta%qble_2".col😆2__11 != 0:::INT8)) OR ("ta%qble_2"."co\vl2_%p_9" >= '+Inf':::FLOAT8)) OR ("ta%qble_2".col2__7 = 1.0:::FLOAT8)) AND ("ta%qble_2".col2__0 < 1:::INT8)) AND ("ta%qble_2".col2__10 < e'\'':::STRING), UNIQUE ("col2_""_1", (CASE WHEN "c?ol2__2" IS NULL THEN 'R':::STRING ELSE e'x)\x0e\x13AU\r':::STRING END) DESC), INDEX (lower(CAST("col2_�_5" AS STRING)) ASC) WHERE (("ta%qble_2".col2__4 <= '\xe29883':::BYTES) OR ("ta%qble_2".col2__12 < e'\U00002603':::STRING)) AND ("ta%qble_2".col2__10 > e'\U00002603':::STRING), INDEX ("co\vl2_%p_9" ASC) STORING ("col2_""_1", col2__8, col😆2__11), UNIQUE ("col2_""_1") WHERE (((((((((("ta%qble_2"."co\vl2_%p_9" = 3.4028234663852886e+38:::FLOAT8) AND ("ta%qble_2".col2__7 <= '+Inf':::FLOAT8)) OR ("ta%qble_2".col2__0 >= 1:::INT8)) AND ("ta%qble_2"."c?ol2__2" = '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ)) OR ("ta%qble_2".col2__12 != 'X':::STRING)) OR ("ta%qble_2".col2__10 > '':::STRING)) OR ("ta%qble_2".col😆2__11 > 1:::INT8)) OR ("ta%qble_2".col2__4 < '\x27':::BYTES)) AND ("ta%qble_2".col2__13 != '"':::STRING)) AND ("ta%qble_2"."col2_""_1" >= e'\'':::STRING)) OR ("ta%qble_2"."col2_�_5" >= '00:00:00+15:59':::TIMETZ), FAMILY ("c o%q😹l2__3", col2__7), FAMILY ("c?ol2__2"), FAMILY ("col%q2__6", col2__4, "col2_""_1"), FAMILY ("col2_�_5", col😆2__11), FAMILY (col2__13, col2__8, col2__0));
    rsg_test.go:720: ALTER TABLE table_1 INJECT STATISTICS e'[{"avg_size": 26, "columns": ["\\"co l1__0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 171316313, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 600000000}, {"avg_size": 15, "columns": ["\\"col1_\\"\\"_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 122687166, "histo_col_type": "", "name": "__auto__", "null_count": 318420709, "row_count": 600000000}, {"avg_size": 20, "columns": ["\\"%vcol1_\\t_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 44481511, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 600000000}, {"avg_size": 28, "columns": ["col1__3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 424402268, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 600000000}]':::JSONB;
    rsg_test.go:720: ALTER TABLE "ta%qble_2" INJECT STATISTICS e'[{"avg_size": 0, "columns": ["col2__4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 18, "histo_col_type": "BYTES", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 10, "columns": ["\\"col2_\\u000c_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 28, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 19, "columns": ["\\"co\\\\vl2_%p_9\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 54, "histo_buckets": [{"distinct_range": 0, "num_eq": 791893605727238656, "num_range": 0, "upper_bound": "0.042747145499876515"}, {"distinct_range": 507.5681649967826, "num_eq": 9000, "num_range": 700, "upper_bound": "0.2193643071164404"}, {"distinct_range": 0, "num_eq": 8000, "num_range": 9000000000, "upper_bound": "0.4319757726488388"}, {"distinct_range": 7000000000, "num_eq": 200000, "num_range": 7000000000, "upper_bound": "-1.1665337124463653"}], "histo_col_type": "FLOAT8", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 0, "columns": ["col2__13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 23, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 3, "columns": ["\\"c o%q\U0001F639l2__3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 53, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 17, "columns": ["col2__12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 12, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 19, "columns": ["col2__10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 81, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 29, "columns": ["\\"col2_\\"\\"_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 55, "histo_buckets": [{"distinct_range": 0, "num_eq": 5000000000, "num_range": 0, "upper_bound": ""}, {"distinct_range": 445.1296101613008, "num_eq": 1043360267306594816, "num_range": 700, "upper_bound": "R"}, {"distinct_range": 0, "num_eq": 2680587205207513999, "num_range": 0, "upper_bound": "p+"}, {"distinct_range": 4801872452935303000, "num_eq": 30000000000, "num_range": 4801872452935302924, "upper_bound": "jwpO"}, {"distinct_range": 795373949.801194, "num_eq": 5694878184141988696, "num_range": 1000000000, "upper_bound": "IG5>p"}, {"distinct_range": 2000366781982070000, "num_eq": 1000000000, "num_range": 6379364448170671740, "upper_bound": "x``\\"kB>"}, {"distinct_range": 0, "num_eq": 2897531853184573821, "num_range": 9000, "upper_bound": "F\\u00118=j\\u000cS"}, {"distinct_range": 70, "num_eq": 800000000, "num_range": 70, "upper_bound": "\\u001f\\u001cn\\"+?j|\\u000b"}], "histo_col_type": "VARCHAR", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 5, "columns": ["\\"c?ol2__2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 16, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 13, "columns": ["col\U0001F6062__11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 40, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 28, "columns": ["col2__0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 71, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 25, "columns": ["col2__7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 11, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 7, "columns": ["col2__8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 57, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 90}, {"avg_size": 27, "columns": ["\\"col%q2__6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 33, "histo_col_type": "", "name": "__auto__", "null_count": 44, "row_count": 90}]':::JSONB;
    rsg_test.go:720: CREATE TYPE rand_typ_0 AS ENUM ('pui');
    rsg_test.go:720: CREATE TYPE rand_typ_1 AS ENUM ('pdbdl', 'loc', 'zpogl', 'gx', 'bilh', 'hl');
    rsg_test.go:720: CREATE TYPE rand_typ_2 AS ENUM ('edh');
    rsg_test.go:720: CREATE TYPE rand_typ_3 AS ENUM ('zp', 'kia');
    rsg_test.go:720: CREATE TYPE rand_typ_4 AS ENUM ('pihyqc');
    rsg_test.go:720: SET sql_safe_updates = false;;
    rsg_test.go:720: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:722: 
    rsg_test.go:723: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith1782519678
--- FAIL: TestRandomSyntaxSQLSmith (380.26s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/json
+++ b/pkg/sql/logictest/testdata/logic_test/json
@@ -1002,3 +1002,15 @@ ORDER BY 1
 1.00
 
 subtest end
+
+# Regression test for not using value-encoding for JSON when fingerprinting a
+# tuple (#118380).
+query T rowsort
+WITH cte1 AS (SELECT * FROM (VALUES
+  (('1':::JSON, 1)),
+  (('2':::JSON, 2))
+) AS tab (col1))
+SELECT * FROM cte1 GROUP BY cte1.col1;
+----
+(1,1)
+(2,2)

--- a/pkg/sql/logictest/testdata/logic_test/tsvector
+++ b/pkg/sql/logictest/testdata/logic_test/tsvector
@@ -435,3 +435,25 @@ true
 # TODO(#75101): The error should be "syntax error in TSQuery".
 statement error pgcode 22023 unsupported comparison operator: <tsvector> @@ <string>
 SELECT 'fat rats'::tsvector @@ 'fat cats chased fat, out of shape rats'
+
+# Regression test for incorrectly using key-encoding on tuples that contain
+# TSQueries and TSVectors (which don't have key-encoding available, #118380).
+query T rowsort
+WITH cte1 AS (SELECT * FROM (VALUES
+  (('1':::TSQUERY, 1)),
+  (('2':::TSQUERY, 2))
+) AS tab (col1))
+SELECT * FROM cte1 GROUP BY cte1.col1;
+----
+('1',1)
+('2',2)
+
+query T rowsort
+WITH cte1 AS (SELECT * FROM (VALUES
+  (('':::TSVECTOR, 1)),
+  (('':::TSVECTOR, 2))
+) AS tab (col1))
+SELECT * FROM cte1 GROUP BY cte1.col1;
+----
+("",1)
+("",2)

--- a/pkg/sql/rowenc/encoded_datum.go
+++ b/pkg/sql/rowenc/encoded_datum.go
@@ -301,6 +301,33 @@ func (ed *EncDatum) Encode(
 	}
 }
 
+func mustUseValueEncodingForFingerprinting(t *types.T) bool {
+	switch t.Family() {
+	// Both TSQuery and TSVector types don't have key-encoding, so we must use
+	// the value encoding for them. JSON type now (as of 23.2) has key-encoding
+	// available, but for historical reasons we will keep on using the
+	// value-encoding (Fingerprint is used by hash routers, so changing its
+	// behavior can result in incorrect results in mixed version clusters).
+	case types.JsonFamily, types.TSQueryFamily, types.TSVectorFamily:
+		return true
+	case types.ArrayFamily:
+		// Note that at time of this writing we don't support arrays of JSON
+		// (tracked via #23468) nor of TSQuery / TSVector types (tracked by
+		// #90886), so technically we don't need to do a recursive call here,
+		// but we choose to be on the safe side, so we do it anyway.
+		return mustUseValueEncodingForFingerprinting(t.ArrayContents())
+	case types.TupleFamily:
+		for _, tupleT := range t.TupleContents() {
+			if mustUseValueEncodingForFingerprinting(tupleT) {
+				return true
+			}
+		}
+		return false
+	default:
+		return false
+	}
+}
+
 // Fingerprint appends a unique hash of ed to the given slice. If datums are intended
 // to be deduplicated or grouped with hashes, this function should be used
 // instead of encode. Additionally, Fingerprint has the property that if the
@@ -321,20 +348,14 @@ func (ed *EncDatum) Fingerprint(
 	var fingerprint []byte
 	var err error
 	memUsageBefore := ed.Size()
-	switch typ.Family() {
-	// Both TSQuery and TSVector types don't have key-encoding, so we must use
-	// the value encoding for them. JSON type now (as of 23.2) has key-encoding
-	// available, but for historical reasons we will keep on using the
-	// value-encoding (Fingerprint is used by hash routers, so changing its
-	// behavior can result in incorrect results in mixed version clusters).
-	case types.JsonFamily, types.TSQueryFamily, types.TSVectorFamily:
+	if mustUseValueEncodingForFingerprinting(typ) {
 		if err = ed.EnsureDecoded(typ, a); err != nil {
 			return nil, err
 		}
 		// We must use value encodings without a column ID even if the EncDatum already
 		// is encoded with the value encoding so that the hashes are indeed unique.
 		fingerprint, err = valueside.Encode(appendTo, valueside.NoColumnID, ed.Datum, nil /* scratch */)
-	default:
+	} else {
 		// For values that are key encodable, using the ascending key.
 		// Note that using a value encoding will not easily work in case when
 		// there already exists the encoded representation because that

